### PR TITLE
xsns_40_pn532.ino small refactoring and add support for NTAG2xx tags

### PR DESCRIPTION
**Pull request by md5sum-as**
## Description: xsns_40_pn532.ino small refactoring and add support for NTAG2xx tags

Added support for NTAG2xx tags (read/write data)
Password protection for NTAG21x (EV1 type tags)
Removed the output of non-printable characters to the console and MQTT.
Removed the check for the presence of non-printable characters when entering a string (they cannot be entered through the console and MQTT)
Changed the format of the Sensor 40 command:
'WRITE,xxx' to write a string (16 characters max) to the card (block 1 for MIFARE Classic and blocks 0x04-0x07 for NTAG).
'ERASE' erase blocks (16).
'AUTH NN,nn' password to access NTAG21x (NN - 32 bit unsigned integer password, nn - 16 bit unsigned integer p_answer)
'SET_PWD' password protect NTAG21x for read and write (SET_PWD NN - for new pass, SET_PWD NN,nn - new pass and answer)
'UNSET_PWD' unprotect NTAG21x

**Question: Is it possible to save the password (uint32_t) and the answer (uint16_t) in the global setting?**

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [x] The code change is tested and works with Tasmota core ESP32 V.2.0.5
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

**Comment by Jason2866** [comment]
Removing data raw is a breaking change. Why is this done?

**Comment by md5sum-as** [comment]
When reading a card with non-printable characters, there is a chance to surprise (confuse) the MQTT broker. 
If there was output via ToHex_P, but it wasn't done
It also doesn't make sense to output non-printable characters to the console.
You won't be able to save non-printable characters from the console - ArgV() is a string function.
I left "sprint()" to check for data from the card. 
Data is written to the card "as is" - a string of characters from the console

File: tasmota/tasmota_xsns_sensor/xsns_40_pn532.ino [hunk 1]
```diff
@@ -393,93 +435,232 @@ uint8_t mifareclassic_WriteDataBlock (uint8_t blockNumber, uint8_t *data) {
   return (0 < PN532_readResponse(Pn532.packetbuffer, sizeof(Pn532.packetbuffer)));
 }
 
+uint8_t ntag21x_probe (void) {
```
Comment (arendst) [inline tasmota/tasmota_xsns_sensor/xsns_40_pn532.ino:438]: Prefix ALL functions with PN532_ as they are global.

Comment (md5sum-as) [reply tasmota/tasmota_xsns_sensor/xsns_40_pn532.ino]: Done. I have added a prefix to ALL the functions

Comment (md5sum-as) [reply tasmota/tasmota_xsns_sensor/xsns_40_pn532.ino]: @arendst, please tell me on the issue of saving the password and the pack in the global configuration (to save it on reboot). Can you do it or should I do it?

Comment (arendst) [reply tasmota/tasmota_xsns_sensor/xsns_40_pn532.ino]: I'll prepare a location in Settings. Once done I let you do the implementation in PN532

Comment (arendst) [reply tasmota/tasmota_xsns_sensor/xsns_40_pn532.ino]: Just added ``Settings->pn532_password`` and ``Settings->pn532_pack`` for you to play with.

**Pull request merged**